trellis/core/time.hpp

```diff
@@ -25,6 +25,7 @@ namespace core {
 namespace time {
 
 using TimePoint = std::chrono::time_point<std::chrono::steady_clock>;
+using SystemTimePoint = std::chrono::time_point<std::chrono::system_clock>;
```

Comment:
Might want to add a note that system_clock doesn't account for leap seconds, so we should be careful with that. I did add the C++20 clock support as an external dep (since we're not using C++20 yet) to get support for gps_clock conversions, so we can use `utc_clock` if instead of system_clock, if we'd like, to be more correct and account for the leap seconds.

Nice summary here: https://stackoverflow.com/questions/64080301/what-are-the-pros-cons-of-the-different-c-clocks-for-logging-time-stamps#:~:text=utc_clock%20is%20just%20like%20system_clock,(or%20a%20fraction%20thereof).

Reply:
I don't anticipate needing use case of subtracting system time points across leap second boundaries. Neither system_time or utc_time is strictly monotonic and thus shouldn't really be used for calculating time intervals for anything critical anyway.